modules/exploits/linux/snmp/net_snmpd_rw_access.rb

```diff
@@ -0,0 +1,130 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+require 'snmp'
+
+class MetasploitModule < Msf::Exploit::Remote
+  Rank = NormalRanking
+
+  include Msf::Exploit::Remote::SNMPClient
+  include Msf::Exploit::EXE
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name'           => 'Net-SNMPd Write Access SNMP-EXTEND-MIB arbitrary code execution',
+        'Description'    => %q(
+            This exploit module exploits the SNMP write access configuration ability of SNMP-EXTEND-MIB to
+ configure MIB extensions and lead to remote code execution.
+        ),
+        'License'        => MSF_LICENSE,
+        'Author'         => ['Steve Embling at InteliSecure'],
+        'References'     =>
+          [
+            [ 'URL', 'https://www.intelisecure.com']
+          ],
+        'Payload'        =>
+          {
+            'Space'    => 4096 #arbitrary, but it seems to max out the size
```

Comment:
This comment is confusing.

Reply:
Thanks, what I meant here is I set it to 20Megs and the payload generated used all the space. So I set it to an arbitrary max size of 4096. Having never touched metasploit source before I'm not sure how this should be handled or what this value should be. Advice appreciated.